src/proposer/proposer_rpc.cpp

```diff
@@ -167,6 +170,125 @@ class ProposerRPCImpl : public ProposerRPC {
     obj.pushKV("stakeable_coins", arr);
     return obj;
   }
+
+  UniValue ProposeBlocks(CWallet &wallet,
+                         const boost::optional<CScript> &coinbase_script,
+                         const int num_generate) const {
+
+    proposer::Proposer *proposer = GetComponent<proposer::Proposer>();
+
+    UniValue block_hashes(UniValue::VARR);
+
+    // To pick up to date coins for staking we need to make sure that the wallet is synced to the current chain.
+    wallet.BlockUntilSyncedToCurrentChain();
+
+    esperanza::WalletExtension &wallet_ext = wallet.GetWalletExtension();
+
+    for (int i = 0; i < num_generate; ++i) {
+
+      std::shared_ptr<const CBlock> block;
+      {
+        LOCK2(cs_main, wallet_ext.GetLock());
+        const staking::CoinSet stakeable_coins = wallet_ext.GetStakeableCoins();
+
+        if (stakeable_coins.empty()) {
+          throw JSONRPCError(RPC_INTERNAL_ERROR,
+                             "Not proposing, not enough balance.");
```

Comment:
NIT: Maybe this can be something like "Not proposing, no stakeable coins available".
"Not enough balance" is a bit confusing to me. I interpret it as "I don't have enough money". And in order to propose I need to get more money.
While in reality there are lots of reasons to not find stakeable coins. For example it might turn out that I need to just wait a bit(for maturity).

Reply:
I share that concern to a certain extent, but:

These RPC are only available when `mine_blocks_on_demand = true`, which is also turning off the eligibility checks for a coin (such that you can always propose). So when in `propose` we do not have coins it really does mean we don't have anything at all, i.e. not enough balance.

Reply:
@scravy I think @AM5800 is right, it might be the case we have coins unlockable with scripts that are not allowed for staking. The balance would be > 0 but they won't be stakeable.

Reply:
@scravy but we still have `IsLockedCoin` and `tx->GetBlocksToRewardMaturity() > 0` checks.

Reply:
True. Then I am in for rewording this too :-)